solr/core/src/java/org/apache/solr/handler/SnapShooter.java

```diff
@@ -104,6 +104,7 @@ private void initialize(
     this.snapshotName = snapshotName;
     if ("file".equals(location.getScheme())) {
       solrCore.getCoreContainer().assertPathAllowed(Paths.get(location));
+      solrCore.getCoreContainer().assertPathNotProtected(Paths.get(location));
```

Comment:
[0] I worry a bit that folks might forget one or the other of these calls.  Maybe it'd be worth refactoring into a single utility method that optionally invokes the not-protected logic in the case of writes.  e.g.

```
public enum AccessType { READ, WRITE, UNKNOWN; }

public void assertPathAllowed(Path toCheck, AccessType access) {
    ...
    if (access == AccessType.WRITE) {
        assertPathNotProtected(...);
    }
}

// Note the 'private' here.
private void assertPathNotProtected(Path toCheck) {
}
```